presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -506,6 +506,26 @@ void PrestoServer::run() {
   addAdditionalPeriodicTasks();
   periodicTaskManager_->start();
 
+  uint64_t systemMemLimitBytes = systemConfig->systemMemLimitGb();
+  uint64_t systemMemShrinkBytes = systemConfig->systemMemShrinkGb();
+#ifdef PRESTO_ENABLE_LINUX_MEMORY_CHECKER
+  linuxMemoryChecker_ =
```

Comment:
Let's pack this block into its own function `startMemoryChecker` and ifdef inside of that function and read of the config variables.